src/shared/sol-conffile.c

```diff
@@ -432,28 +450,113 @@ _fill_vector(const char *path, const char *fallback_paths)
     if (_json_to_vector(json_scanner) != 0)
         goto free_for_all;
 
-    if (include || include_fallbacks)
-        _fill_vector(include, include_fallbacks);
+    if (include || include_fallbacks.len) {
+        _fill_vector(include, &include_fallbacks);
+    }
 
 free_for_all:
     sol_file_reader_close(file_reader);
     free(full_path);
     free(include);
-    free(include_fallbacks);
+
+    SOL_VECTOR_FOREACH_IDX (&include_fallbacks, slice, i) {
+        free((char *)slice->data);
+    }
+
+    sol_vector_clear(&include_fallbacks);
+}
+
+static void
+_add_lookup_path(struct sol_vector *vector, const char **search_dirs, char *appname, const char *board_name)
+{
+    size_t i;
+    struct sol_vector files = SOL_VECTOR_INIT(struct sol_str_slice);
+
+#define ADD_PATH_FMT(_vector, _fmt, ...) \
+    do { \
+        int _r; \
+        char *_buff; \
+        struct sol_str_slice *_s; \
+        _r = asprintf(&_buff, _fmt, ## __VA_ARGS__); \
+        if (_r < 0 || !_buff) \
+            SOL_WRN("Couldn't allocate memory for config file formatting."); \
+        else { \
+            _s = sol_vector_append(_vector); \
+            if (_s) { \
+                _s->data = _buff; \
+                _s->len = strlen(_buff); \
+            } else { \
+                free(_buff); \
+            } \
+        } \
+    } while (0) \
```

Comment:
Stray `\`. Please make this a function, there's no need to make it a macro. (Also, it can be reused above.)